uaclient/cli/fix.py

```diff
@@ -52,6 +104,747 @@ def fix_parser(parser):
+            print(
+                "- "
+                + messages.ENABLE_REBOOT_REQUIRED_TMPL.format(
+                    operation="fix operation"
+                )
+            )
+            failure_on_related_usn = True
+        if status == FixStatus.SYSTEM_STILL_VULNERABLE:
+            for unfixed_pkg in unfixed_pkgs:
+                if unfixed_pkg.unfixed_reason:
+                    print(
+                        "  - {}: {}".format(
+                            unfixed_pkg.pkg, unfixed_pkg.unfixed_reason
+                        )
+                    )
+            failure_on_related_usn = True
+
+    if failure_on_related_usn:
+        print(
+            "\n"
+            + messages.SECURITY_RELATED_USN_ERROR.format(
+                issue_id=security_issue
+            )
+        )
+
+    return target_usn_status
+
+
+def status_message(status, pocket_source: Optional[str] = None):
+    if status == "needed":
+        return messages.SECURITY_CVE_STATUS_NEEDED
+    elif status == "needs-triage":
+        return messages.SECURITY_CVE_STATUS_TRIAGE
+    elif status == "pending":
+        return messages.SECURITY_CVE_STATUS_PENDING
+    elif status in ("ignored", "deferred"):
+        return messages.SECURITY_CVE_STATUS_IGNORED
+    elif status == "DNE":
+        return messages.SECURITY_CVE_STATUS_DNE
+    elif status == "not-affected":
+        return messages.SECURITY_CVE_STATUS_NOT_AFFECTED
+    elif status == "released" and pocket_source:
+        return messages.SECURITY_FIX_RELEASE_STREAM.format(
+            fix_stream=pocket_source
+        )
+    return messages.SECURITY_CVE_STATUS_UNKNOWN.format(status=status)
+
+
+def _format_packages_message(
+    pkg_list: List[str],
+    status: str,
+    pkg_index: int,
+    num_pkgs: int,
+    pocket_source: Optional[str] = None,
+) -> str:
+    """Format the packages and status to an user friendly message."""
+    if not pkg_list:
+        return ""
+
+    msg_index = []
+    src_pkgs = []
+    for src_pkg in pkg_list:
+        pkg_index += 1
+        msg_index.append("{}/{}".format(pkg_index, num_pkgs))
+        src_pkgs.append(src_pkg)
+
+    msg_header = textwrap.fill(
+        "{} {}:".format(
+            "(" + ", ".join(msg_index) + ")", ", ".join(sorted(src_pkgs))
+        ),
+        width=PRINT_WRAP_WIDTH,
+        subsequent_indent="    ",
+    )
+    return "{}\n{}".format(msg_header, status_message(status, pocket_source))
+
+
+def _run_ua_attach(cfg: UAConfig, token: str) -> bool:
+    """Attach to an Ubuntu Pro subscription with a given token.
+
+    :return: True if attach performed without errors.
+    """
+    import argparse
+
+    from uaclient import cli
+
+    print(colorize_commands([["pro", "attach", token]]))
+    try:
+        ret_code = cli.action_attach(
+            argparse.Namespace(
+                token=token, auto_enable=True, format="cli", attach_config=None
+            ),
+            cfg,
+        )
+        return ret_code == 0
```

Comment:
Now might be a good time to change this to use `actions.attach_with_token` instead